databricks/provider.go

```diff
@@ -209,6 +265,11 @@ func providerConfigure(d *schema.ResourceData, providerVersion string) (interfac
 		if token, ok := d.GetOk("token"); ok {
 			config.Token = token.(string)
 		}
+		if config.Host == "" || config.Token == "" {
+			if err := tryDatabricksCliConfigFile(d, &config); err != nil {
+				return nil, err
+			}
+		}
```

Comment:
I'd probaly go with something like this for separation of concerns and readability:
```
if config.Host == "" || config.Token == "" {
  pathRaw, ok := d.GetOk("config_file")
  if !ok { return err... }
  path, ok  := pathRaw.(string) // Not sure if ok check is needed here. I see it's never done, so maybe it doesn't panic if d.GetOk returns ok = true
  if !ok {
    return fmt.errorf(...)
  }
  profile, ok := d.Get("profile") 
... // similar to path

  host, token, err := getCredentials(path, profile) // this function would open file and try to read profile settings.
  if err {...}
  config.Host = host
  config.Token = token
}
```